lib/src/log_handler.dart

```diff
@@ -0,0 +1,225 @@
+  final String flutterVersion;
+  final String dartVersion;
+  final String tool;
+  final DateTime localTime;
+
+  LogItem({
+    required this.sessionId,
+    required this.branch,
+    required this.host,
+    required this.flutterVersion,
+    required this.dartVersion,
+    required this.tool,
+    required this.localTime,
+  });
+
+  /// Serves a parser for each record in the log file
+  ///
+  /// Using this method guarantees that we have parsed out
+  /// fields that are necessary for the [LogHandler.logFileStats]
+  /// method
+  ///
+  /// If the returned value is [null], that indicates a malformed
+  /// record which can be discarded during analysis
+  ///
+  /// Example of what a record looks like:
+  /// ```
+  /// {
+  ///     "client_id": "d40133a0-7ea6-4347-b668-ffae94bb8774",
+  ///     "events": [
+  ///         {
+  ///             "name": "hot_reload_time",
+  ///             "params": {
+  ///                 "time_ns": 345
+  ///             }
+  ///         }
+  ///     ],
+  ///     "user_properties": {
+  ///         "session_id": {
+  ///             "value": 1675193534342
+  ///         },
+  ///         "branch": {
+  ///             "value": "ey-test-branch"
+  ///         },
+  ///         "host": {
+  ///             "value": "macOS"
+  ///         },
+  ///         "flutter_version": {
+  ///             "value": "Flutter 3.6.0-7.0.pre.47"
+  ///         },
+  ///         "dart_version": {
+  ///             "value": "Dart 2.19.0"
+  ///         },
+  ///         "tool": {
+  ///             "value": "flutter-tools"
+  ///         },
+  ///         "local_time": {
+  ///             "value": "2023-01-31 14:32:14.592898"
+  ///         }
+  ///     }
+  /// }
+  /// ```
+  static LogItem? fromRecord(Map<String, dynamic> record) {
+    if (!record.containsKey('user_properties')) return null;
+
+    // Using a try/except here to parse out the fields if possible,
+    // if not, it will quietly return null and won't get processed
+    // downstream
+    try {
+      // Parse the data out of the `user_properties` value
+      final Map<String, dynamic> userProps = record['user_properties'];
+
+      // Parse out the values from the top level key = 'user_properties`
+      final int sessionId = userProps['session_id']['value'];
+      final String branch = userProps['branch']['value'];
+      final String host = userProps['host']['value'];
+      final String flutterVersion = userProps['flutter_version']['value'];
+      final String dartVersion = userProps['dart_version']['value'];
+      final String tool = userProps['tool']['value'];
+      final DateTime localTime =
+          DateTime.parse(userProps['local_time']['value']);
+
+      return LogItem(
+        sessionId: sessionId,
+        branch: branch,
+        host: host,
+        flutterVersion: flutterVersion,
+        dartVersion: dartVersion,
+        tool: tool,
+        localTime: localTime,
+      );
+    } catch (e) {
+      return null;
+    }
+  }
```

Comment:
@christopherfujino let me know what you think about this approach – this will just return null for any malformed records

Reply:
I think this will stop working if you change the parameter type from `dynamic` -> `Object?`, per the style guide. Also, it's not safe to catch everything, per this lint (which we have enabled in the tool) https://dart-lang.github.io/linter/lints/avoid_catches_without_on_clauses.html.

I know it's more verbose, but I think you should remove the dynamic and explicitly check for nulls.

Reply:
That's fair, I've removed all instances of `dynamic` from the package code and cast them as the appropriate nullable type now